### PR TITLE
feat(routing): add dynamic routing and basic page navigation

### PR DESCRIPTION
**Pull request by mahdisolgi1**
- Enable navigation to empty pages: Home, Blog, and Posts/{post-slug}
- Implement dynamic routing based on environment variables (e.g., dev-tools route)

### CODE REVIEW

File: frontend/src/pages/Root.tsx
Comment (amirrezaDev1378) [inline frontend/src/pages/Root.tsx:1]: add a css class for styling instead of inline styling
@mahdisolgi1

**Comment by amirrezaDev1378** [comment]
I suggest to use module css:
https://vite.dev/guide/features#css-modules

**Comment by amirrezaDev1378** [comment]
Please apply types for all components and their props (if they have any)

**Comment by amirrezaDev1378** [comment]
check this out
https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/basic_type_example

**Comment by mahdisolgi1** [comment]
I made some changes in commits ca99293 and 389b476 that address the issues .

**Comment by mahdisolgi1** [comment]
If there’s anything more that can be implemented, please feel free to share

**Review by amirrezaDev1378 — APPROVED** [review]
LGTM!

**Pull request merged**